### PR TITLE
UI Test Cleanup: Split out out_of_bounds_indexing

### PR DESCRIPTION
**Pull request by phansch**
This moves the `out_of_bounds_indexing` lint tests to their own
directory.

changelog: none

cc #2038

### CODE REVIEW

File: tests/ui/out_of_bounds_indexing/empty_array.stderr [hunk 1]
```diff
@@ -0,0 +1,54 @@
+error: index out of bounds: the len is 0 but the index is 0
+  --> $DIR/empty_array.rs:6:5
+   |
+LL |     empty[0]; // Ok, let rustc's `const_err` lint handle `usize` indexing on arrays.
```
Comment (flip1995) [inline tests/ui/out_of_bounds_indexing/empty_array.stderr]: I think emitting the lint here is fine, but could you remove the comment?

File: tests/ui/out_of_bounds_indexing/issue-3102.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#![warn(clippy::out_of_bounds_indexing)]
```
Comment (flip1995) [inline tests/ui/out_of_bounds_indexing/issue-3102.rs]:
```suggestion
#![warn(clippy::out_of_bounds_indexing)]
#![allow(clippy::no_effect)]
```

File: tests/ui/out_of_bounds_indexing/empty_array.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#![warn(clippy::out_of_bounds_indexing)]
+#![allow(clippy::no_effect, clippy::unnecessary_operation)]
+
+fn main() {
+    let empty: [i8; 0] = [];
+    empty[0]; // Ok, let rustc's `const_err` lint handle `usize` indexing on arrays.
```
Comment (flip1995) [inline tests/ui/out_of_bounds_indexing/empty_array.rs]:
```suggestion
    empty[0];
```

**Comment by flip1995** [comment]
`./util/dev fmt`

File: tests/ui/out_of_bounds_indexing/empty_array.stderr (hunk 1 above)
Comment (phansch) [reply tests/ui/out_of_bounds_indexing/empty_array.stderr]: Done :+1:

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Pull request merged**